common/Util.cpp

```diff
@@ -113,43 +113,44 @@ namespace Util
             return _rng();
         }
 
+        int getURandom()
+        {
+            static int urandom = open("/dev/urandom", O_RDONLY);
+            if (urandom < 0)
+            {
+                LOG_SYS("Failed to source hard random numbers");
+                fprintf(stderr, "No adequate source of randomness");
+                abort();
+                // Potentially dangerous to continue without randomness
+            }
+            return urandom;
+        }
+
+        // Since we have a fd always open to /dev/urandom
+        // 'read' is hopefully no less efficient than getrandom.
         std::vector<char> getBytes(const std::size_t length)
         {
             std::vector<char> v(length);
 
-            int len = 0;
-            /* use "/dev/[u]random" by default in case of
-            getentropy() or getrandom() fails,
-            or they don't available */
-            bool useDevRandom = true;
-#ifdef HAVE_GETENTROPY
```

Comment:
Then we can drop the HAVE_SYS_RANDOM_H and HAVE_GETENTROPY include guard above, the configure check for those and the config.h.in mentions

Reply:
Yes - I believe so - of course we need the core commits before this one will pass too of course.